pkg/k8s/watchers/secret.go

```diff
@@ -31,62 +31,64 @@ const (
 	caCrtAttribute = "ca.crt"
 )
 
-func (k *K8sWatcher) tlsSecretInit(slimClient slimclientset.Interface, namespace string, swgSecrets *lock.StoppableWaitGroup) {
+func (k *K8sWatcher) tlsSecretInit(slimClient slimclientset.Interface, namespaces []string, swgSecrets *lock.StoppableWaitGroup) {
 	// Watch for all Secret types
 	secretOptsModifier := func(options *metav1.ListOptions) {}
 
 	apiGroup := resources.K8sAPIGroupSecretV1Core
-	_, secretController := informer.NewInformer(
-		utils.ListerWatcherWithModifier(
-			utils.ListerWatcherFromTyped[*slim_corev1.SecretList](slimClient.CoreV1().Secrets(namespace)),
-			secretOptsModifier),
-		&slim_corev1.Secret{},
-		0,
-		cache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj interface{}) {
-				var valid, equal bool
-				defer func() {
-					k.K8sEventReceived(apiGroup, metricSecret, resources.MetricCreate, valid, equal)
-				}()
-				if k8sSecret := k8s.ObjToV1Secret(obj); k8sSecret != nil {
-					valid = true
-					err := k.addK8sSecretV1(k8sSecret)
-					k.K8sEventProcessed(metricSecret, resources.MetricCreate, err == nil)
-				}
-			},
-			UpdateFunc: func(oldObj, newObj interface{}) {
-				var valid, equal bool
-				defer func() { k.K8sEventReceived(apiGroup, metricSecret, resources.MetricUpdate, valid, equal) }()
-				if oldSecret := k8s.ObjToV1Secret(oldObj); oldSecret != nil {
-					if newSecret := k8s.ObjToV1Secret(newObj); newSecret != nil {
+	for _, ns := range uniq(namespaces) {
+		_, secretController := informer.NewInformer(
+			utils.ListerWatcherWithModifier(
+				utils.ListerWatcherFromTyped[*slim_corev1.SecretList](slimClient.CoreV1().Secrets(ns)),
+				secretOptsModifier),
+			&slim_corev1.Secret{},
+			0,
+			cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					var valid, equal bool
+					defer func() {
+						k.K8sEventReceived(apiGroup, metricSecret, resources.MetricCreate, valid, equal)
+					}()
```

Comment:
NIt: we can avoid the creation of an anonymous function and directly defer the call to `k.K8sEventReceived(...)`.
Same in `UpdateFunc` and `DeleteFunc` callback

Reply:
ah actually, the anonymous function is to make sure `valid` variable is not evaluated right away.

Reply:
That's right, I was missing the closure reference to `valid`. Thank you for the explanation. :pray: